### ISSUE TITLE
Amiberry runs stable on Android (Termux/X11/XFCE4) but the executable only runs from parent of build dir..

### ISSUE THREAD

**Issue by PTz0uAH** [BlitterStudio/amiberry#1490]
Hoi @midwan behold Amiberry on Android.. (non-root)

![amiberry-aarch64-termuxified1](https://github.com/user-attachments/assets/d6ad82b5-0dd2-4867-b48d-33716589e7d3)

Hacked the code a bit to make it work.. committed to the fork..

Maybe an interesting new target for Amiberry..

with kind regards,

Peter

**Comment by giantclambake** [comment]
That would be expected... see -> https://github.com/BlitterStudio/amiberry/wiki/Amiberry-directories

Follow that up with  --> https://github.com/tox-dev/platformdirs/issues/70

**Comment by midwan** [comment]
@PTz0uAH 
Thanks for that!
From a quick look, I think the hacked changes you made could be better incorporated into the main code, perhaps with some checks. 

I'm not familiar with Termux, but there have been some attempts to bring Amiberry to Android in the past. We never got it further than a first alpha version, and eventually the person helping with this disappeared, so the effort was never completed. I think bringing it to Android in general would be interesting, even if it will need some further changes (like on-screen controls).

**Comment by PTz0uAH** [comment]
Well, I run Termux-app which is providing a Linux desktop as an Android app, so the path logic is non-android.. Amiberry is compiled as a normal desktop program yet needs to be a position-independent-executable.. which needs recompilation..

The usecase for me would be creating on-device Amiga environments to also be used with the great Uae4arm which runs in parallel on the "normal" app-side.. which is amazing.. for me as Oldschool Amiga fan..

**Comment by PTz0uAH** [comment]
@midwan

I want to make the hacks/edits more complient to master-origin so I need to introduce a `{$define TERMUX}`.. what unit would be best to create such a global define?

**Comment by midwan** [comment]
@PTz0uAH 
Yeah, after checking out Termux a bit more, it's more like running Amiberry on a Linux sandbox, on Android. :)
I assume you had to install X11 and XFCE yourself inside Termux first?

It might be easier to just install the deb/rpm package inside that then, instead of doing a manual build - of course, that would be possible after the fixes are included, and a package is generated.

The `TERMUX` define would be one way, which could be defined in the main `CMakeLists.txt` file as an option (there are already a few there, as you can see). I think at least some of those changes could be changed however, to check for something in the system instead, and not special-case everything with `#ifdef TERMUX`.

**Comment by midwan** [comment]
Building with `pie` enabled is not an issue (though JIT will then not work), the macOS builds do that already, since that's forced in the compiler. We can use a CMake option for it, I guess.

Release builds should not be that large, and CPack should strip them before packaging anyway. If Termux can use DEB packages, but it has a different directory structure for example, it would be best to use the CPack options already available and perhaps use a different PREFIX value? Not sure what else is different there.

Alternatively, maybe a Flatpak can be used in Termux? There's already one on Flathub, and it will be updated with the new release as soon as it's ready.

**Comment by midwan** [comment]
Very nice! It will be interesting to see if we can make this work in a sort of standardized way, so that new releases include this package as well.
